### ISSUE TITLE
share scan crash (size/levels, unicode)

### ISSUE THREAD

**Issue by dchmelik** [nicotine-plus/nicotine-plus#2000]
Nicotine+ version: nicotine+-3.0.0-x86_64-1_SBo
Operating System/Distribution: Slackware64 15.1-current GNU/Linux

### Describe the bug
Nicotine+ share scan crashes: says I have zero, and has unicode problems.

### Expected behavior
Scan many GB shares.  Some peer-to-peer filesharers have 100+TB (probably more classification detail--more directory levels--than me) so possible size including directory levels needs increase, and should support more unicode.

### Steps to reproduce the bug
Start Nicotine+ with large/detailed share, many directory levels, and unicode characters in file names you downloaded.

### Additional context
partial log:
```
20:47:03 Error while scanning folder /home/d/share/math/src/games/roguelike/moria_angband_tome: [Errno 40] Too many levels of symbolic links: '/home/d/share/math/src/games/roguelike/moria_angband_tome/moria_angband_tome'
20:47:03 Error while scanning folder /home/d/share/math/src/games/roguelike/angband: [Errno 40] Too many levels of symbolic links: '/home/d/share/math/src/games/roguelike/angband/moria_angband_tome'
20:47:03 Error while scanning folder /home/d/share/math/src/games/roguelike/tome: [Errno 40] Too many levels of symbolic links: '/home/d/share/math/src/games/roguelike/tome/moria_angband_tome'
20:47:03 Error while scanning folder /home/d/share/math/src/games/roguelike/moria: [Errno 40] Too many levels of symbolic links: '/home/d/share/math/src/games/roguelike/moria/moria_angband_tome'
20:47:04 Serious error occurred while rescanning shares. If this problem persists, delete /home/d/.local/share/nicotine/*.db and try again. If that doesn't help, please file a bug report with this stack trace included: 
Traceback (most recent call last):
  File "/usr/lib64/python3.9/site-packages/pynicotine/shares.py", line 103, in run
    self.rescan_dirs(
  File "/usr/lib64/python3.9/site-packages/pynicotine/shares.py", line 185, in rescan_dirs
    newsharedfiles, newsharedfilesstreams = self.get_files_list(newmtimes, oldmtimes, oldfiles, oldstreams, rebuild)
  File "/usr/lib64/python3.9/site-packages/pynicotine/shares.py", line 292, in get_files_list
    if not rebuild and folder in oldmtimes:
  File "/usr/lib64/python3.9/shelve.py", line 102, in __contains__
    return key.encode(self.keyencoding) in self.dict
UnicodeEncodeError: 'utf-8' codec can't encode character '\udcbd' in position 106: surrogates not allowed
```

**Comment by mathiascode** [comment]
You seem to be using Nicotine+ 3.0.0, which is over a year old at this point. There have been fixes for issues like these since. Could you try the latest version (3.2.2) and see if you still experience issues (and perhaps ask the maintainer of the Slackware package if they could update the package)?

**Comment by dchmelik** [comment]
I increased version number in SlackBuild to upgrade to 3.2.2, still get such errors but also many 'no such file or directory,' 'error while scanning metadata'... it is scanning now but will take a while maybe with more errors.

**Comment by slook** [comment]
Large fixes were done since version 3.2.0 which makes the scanner much more tolerant to errors, and also with improved logging of errors so it is to be expected that you will see more errors than before, but the errors are better handled rather than crashing the scanner module. Hopefully the scan will complete and most of your library will be indexed.

Your continued testing with the very large and complex shares library would be very much appreciated... Please continue to post the logged error messages for those items which for some reason cannot be indexed.
